common/lib/xmodule/xmodule/discussion_module.py

```diff
@@ -42,7 +44,20 @@ class DiscussionFields(object):
     sort_key = String(scope=Scope.settings)
 
 
+def has_permission(user, permission, course_id):
+    """
+    Copied from django_comment_client/permissions.py because I can't import
+    that file from here. It causes the xmodule_assets command to fail.
+    """
+    return any(role.has_permission(permission)
+               for role in user.roles.filter(course_id=course_id))
+
+
+@XBlock.wants('user')
```

Comment:
As I understand it, "wants" is for when your XBlock can use the service if it's available, or can fallback on something else if it's not available. "needs" is for when your XBlock can't function if the service is unavailable.